### PR TITLE
Stripe payment issue fixed

### PR DESCRIPTION
**Pull request by sowmiyasubramaniyan**
Run the following command:
php artisan migrate
php artisan db:seed --class="Database\Seeders\v4_0_2\DatabaseSeeder"

### CODE REVIEW

File: app/Http/Controllers/RazorpayController.php [hunk 1]
```diff
@@ -220,4 +219,37 @@ private function doTheDeed($invoice)
...
+        try {
+            $stripeSecretKey = ApiKey::pluck('stripe_secret')->first();
+            $stripe = new \Stripe\StripeClient($stripeSecretKey);
+            $invoice = \Session::get('invoice');
+            $paymentIntent = $stripe->paymentIntents->retrieve($request->input('payment_intent'));
+            if ($paymentIntent->status === 'succeeded') {
+                $currency = strtolower($invoice->currency);
+                $controller = new SettingsController();
+                $result = $controller->processPaymentSuccess($invoice, $currency);
+                \Session::forget('items');
```
Comment (RafficMohammed) [inline app/Http/Controllers/RazorpayController.php:234]: \Session::forget([]) => mention all the keys here

**Review by RafficMohammed — CHANGES_REQUESTED** [review]

**Review by RafficMohammed — APPROVED** [review]

**Review by sandesh556 — DISMISSED** [review]

**Comment by sowmiyasubramaniyan** [comment]
I am putting this on hold for now, as I am fixing and pushing the Razorpay auto-renewal subscription fixes in this same PR. Also, I need to do code cleanup for the payment functions.

**Review by SarfrazS — CHANGES_REQUESTED** [review]
- [ ] Issue1:- In Auto reminder mail still price showing wrong. Yesterday you told me i fixed it. Please check and let me know.
image

- [ ] Issue2:-Auto payments are failed with 3D card.
image

- [ ] Issue3:- Unable to test cloud products with 2days plan. in a group showing yearly. but that is also not able to be purchased.
image

**Review by SarfrazS — CHANGES_REQUESTED** [review]
- [ ] Issue:- User received start subscription mail but short code, Total amount and make payment button not working.
Screenshot 2024-03-22 103017

- [ ] When we are editing subscription template getting below error.
image

**Review by SarfrazS — CHANGES_REQUESTED** [review]
- [ ] The first time subscription mail was received and I made the manual payment, the second time should auto payment but not happen order expired.

- [ ] I purchased a cloud product with Stripe not receiving subscription mail.
- [ ] I purchased a self-hosted product with Razorpay, but still didn't receive subscription mail.
- [ ] I purchased self-hosted products with USD from Razorpay test card. Auto renewal is enabled but still auto-payment mail is not received.

**Review by sandesh556 — APPROVED** [review]

**Review by SarfrazS — APPROVED** [review]
Tested the below scenarios with Razor pay from Domestic and international cards.
•	Purchases self-hosted products with Razorpay INR currency.
•	Purchases self-hosted products with Razorpay USD currency.
•	Checked purchase confirmation mail received on mail.
•	Checked invoice mail received on mail.
•	Checked order expiry mail received on mail.
•	Checked reminder mail received on mail.
•	Checked manual renewal with INR currency.
•	Check manual renewal with USD currency.
•	Enabled auto-renewal and checked auto-renewal with INR currency.
•	Checked first-time INR currency users subscribe manually after that auto-renewal will be happening.
•	Checked once the INR currency subscription got active then the auto-renewal billing cycle working.
•	Checked after subscription got active next due payment auto payment happening. 
•	Checked subscription mail session expiry, after 24 hours subscription link expired.
•	Checked after the subscription link expired when the user clicked on it showed void. (Message showing from payment gateway)
•	 Enabled auto-renewal and checked auto-renewal with USD currency.
•	Checked for USD currency subscription mail not sending directly autopayment happening.
•	Checked INR currency users can't pay with USD cards.
•	Checked USD currency users can’t pay with INR cards.
•	Checked from which card auto-renewal active from that payment gateway subscription got initialized.
•	Checked country-based plan price auto-renewal.
•	Checked if the country-based plan is not there then picked default price in auto-renewal.
•	Checked failed payment mail received on mail.
•	Purchases Cloud products with Razorpay INR currency.
•	Purchases Cloud products with Razorpay USD currency.
•	Checked Enabled Auto-renewal with INR currency.
•	Checked Enabled Auto-renewal with USD currency.
•	Without agents increase/decrease checked auto-renewal with INR.
•	Without agents increase/decrease checked auto-renewal with INR.
•	First increased Agents and checked auto-renewal with INR.
•	First Increased Agents and checked auto-renewal with USD.
•	First decreased Agents and checked auto-renewal with INR.
•	First decreased Agents and checked auto-renewal with USD.
•	Checked auto-renewal after products upgraded from HDM to SDM with INR.
•	Checked auto-renewal after products upgraded from HDM to SDM with USD.
•	Checked auto-renewal after products were downgraded from HDM to SDM with INR.
•	Checked auto-renewal after products were downgraded from HDM to SDM with USD.

Tested the below scenarios with Stripe from 3DS and non-3DS Cards as well as individual country cards.
•	Purchases self-hosted products with Stripe INR currency.
•	Purchases self-hosted products with Stripe USD currency.
•	Checked purchase confirmation mail received on mail.
•	Checked invoice mail received on mail.
•	Checked order expiry mail received on mail.
•	Checked reminder mail received on mail.
•	Checked manual renewal with INR currency.
•	Check manual renewal with USD currency.
•	Enabled auto-renewal and checked auto-renewal with INR currency.
•	Checked first-time INR currency users subscribe manually after that auto-renewal will be happening.
•	Checked once the INR currency subscription got active then the auto-renewal billing cycle working.
•	Checked after subscription got active next due payment auto payment happening. 
•	Checked subscription mail session expiry, after 24 hours subscription link expired.
•	Checked after the subscription link expired when the user clicked on it showed void. (Message showing from payment gateway)
•	 Enabled auto-renewal and checked auto-renewal with USD currency.
•	Checked for USD currency subscription mail not sending directly autopayment happening.
•	Checked INR currency users can't pay with USD cards.
•	Checked USD currency users can’t pay with INR cards.
•	Checked from which card auto-renewal active from that payment gateway subscription got initialized.
•	Checked country-based plan price auto-renewal.
•	Checked if the country-based plan is not there then picked default price in auto-renewal.
•	Checked failed payment mail received on mail.
•	Purchases Cloud products with Stripe INR currency.
•	Purchases Cloud products with Stripe USD currency.
•	Checked Enabled Auto-renewal with INR currency.
•	Checked Enabled Auto-renewal with USD currency.
•	Without agents increase/decrease checked auto-renewal with INR.
•	Without agents increase/decrease checked auto-renewal with INR.
•	First increased Agents and checked auto-renewal with INR.
•	First Increased Agents and checked auto-renewal with USD.
•	First decreased Agents and checked auto-renewal with INR.
•	First decreased Agents and checked auto-renewal with USD.
•	Checked auto-renewal after products upgraded from HDM to SDM with INR.
•	Checked auto-renewal after products upgraded from HDM to SDM with USD.
•	Checked auto-renewal after products were downgraded from HDM to SDM with INR.
•	Checked auto-renewal after products were downgraded from HDM to SDM with USD.
image

**Pull request merged**